flang/lib/Lower/OpenMP.cpp

```diff
@@ -397,6 +403,19 @@ getSIMDModifier(const Fortran::parser::OmpScheduleClause &x) {
   return mlir::omp::ScheduleModifier::none;
 }
 
+int64_t Fortran::lower::getCollapseValue(
+    const Fortran::parser::OmpClauseList &clauseList) {
+  int64_t collapseValue = 1;
+  for (const auto &clause : clauseList.v) {
+    if (const auto &collapseClause =
+            std::get_if<Fortran::parser::OmpClause::Collapse>(&clause.u)) {
+      const auto *expr = Fortran::semantics::GetExpr(collapseClause->v);
+      collapseValue = Fortran::evaluate::ToInt64(*expr).value();
```

Comment:
Why not doing an early return when collapse is found. There should be no more than one collapse clause in the list anyway after semantic.

Reply:
Finally got round to do this.